components/authz-service/storage/postgres/migration/legacy/iam.go

```diff
@@ -1,45 +1,73 @@
-package v2
+package legacy
 
 import (
 	"context"
+	"database/sql"
 	"fmt"
 	"strings"
 
-	"github.com/pkg/errors"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
-	constants_v1 "github.com/chef/automate/components/authz-service/constants/v1"
-	constants_v2 "github.com/chef/automate/components/authz-service/constants/v2"
 	storage_errors "github.com/chef/automate/components/authz-service/storage"
-	storage_v1 "github.com/chef/automate/components/authz-service/storage/v1"
-	storage "github.com/chef/automate/components/authz-service/storage/v2"
+	constants_v1 "github.com/chef/automate/components/authz-service/storage/postgres/migration/legacy/constants/v1"
+	constants_v2 "github.com/chef/automate/components/authz-service/storage/postgres/migration/legacy/constants/v2"
+	"github.com/chef/automate/lib/logger"
 	uuid "github.com/chef/automate/lib/uuid4"
+	"github.com/pkg/errors"
 )
 
-// Note: it would be silly to translate the migration status into errors twice,
-// so this method is an exception to the "only grpc handlers return status
-// instances" rule.
-func (s *policyServer) okToMigrate(ctx context.Context, ms storage.MigrationStatus) error {
-	switch ms {
-	case storage.Successful, storage.SuccessfulBeta1:
-		return status.Error(codes.AlreadyExists, "already migrated")
-	case storage.InProgress:
-		return status.Error(codes.FailedPrecondition, "migration already in progress")
-	case storage.Failed:
-		s.log.Info("migration failed before; attempting again")
+// MigrateToV2 inserts needed IAM v2 resources into the db and
+// migrates any valid v1 policies
+func MigrateToV2(ctx context.Context, db *sql.DB) error {
```

Comment:
This is much of the logic from the original MigrateToV2 call-- called from our migration file when postgres is initialized